src/providers/pass-certificate-validator/pass-certificate-validator.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class PassCertificateValidatorProvider {
+
+  isPassCertificateValid(certificate: string | null): boolean {
+    if (!certificate || certificate.length !== 8) {
+      return false;
+    }
+    if (!this.isLetter(certificate[0])) {
+      return false;
+    }
+
+    const checkDigit = this.calculateMod37CheckDigit(certificate);
+
+    if (checkDigit === 'invalid') {
+      return false;
+    }
+
+        // compare entered checkdigit with generated one
+    if (checkDigit !== certificate.toUpperCase()[7]) {
```

Comment:
Would apply `toUpperCase` to `checkDigit` as well, just to make sure that both are uppercase in all cases.

No need for the comment IMO

Reply:
Done